cartographer_rviz/cartographer_rviz/drawable_submap.cc

```diff
@@ -159,13 +168,16 @@ void DrawableSubmap::SetAlpha(const double current_tracking_z) {
 
 void DrawableSubmap::UpdateSceneNode() {
   ::cartographer::common::MutexLocker locker(&mutex_);
+  render_in_progress_ = false;
+  if (quitting_) {
```

Comment:
I'd say you are overthinking it, and I don't think your suggestion would make it safer.

If it were possible for the outside to call a public member function (e.g. `MaybeFetchTexture()` as you said) and call the destructor before this function call finishes, I don't think you can have correct behavior without keeping some data outside of the object. The destructor might have completed before the called function starts executing its first line in a different thread, and would try to take a mutex that has already been destroyed. In short: we depend on the destructor only being called when there are no outstanding calls to any of the public member functions, and I think this is correct for us to assume.